core/src/components/modal/test/sheet/e2e.ts

```diff
@@ -117,3 +117,86 @@ test('input should not be focusable when backdrop is active', async () => {
   const parentEl = await getActiveElement(page);
   expect(parentEl.tagName).toEqual('ION-BUTTON');
 });
+
+describe('modal: sheet: setting the breakpoint', () => {
+
+  let page: E2EPage;
+
+  beforeEach(async () => {
+    page = await newE2EPage({
+      url: '/src/components/modal/test/sheet?ionic:_testing=true'
+    });
+  });
+
+  describe('setting an invalid value', () => {
+
+    let warnings: string[];
+    let modal: E2EElement;
+
+    beforeEach(async () => {
+      warnings = [];
+      page.on('console', (ev) => {
+        if (ev.type() === 'warning') {
+          warnings.push(ev.text());
+        }
+      });
+
+      modal = await openModal(page, '#sheet-modal');
+
+      await modal.callMethod('setCurrentBreakpoint', 0.01);
+      await page.waitForChanges();
+    });
+
+    it('should not change the breakpoint', async () => {
+      const breakpoint = await modal.callMethod('getCurrentBreakpoint');
+      expect(breakpoint).toBe(0.25);
+    });
+
+    it('should console a warning to developers', async () => {
+      expect(warnings.length).toBe(1);
+      expect(warnings[0]).toBe('[Ionic Warning]: Attempted to set invalid breakpoint value 0.01. Please double check that the breakpoint value is part of your defined breakpoints.');
+    });
+
+  });
+
+  describe('setting the breakpoint to the same value', () => {
+
+    it('should do nothing', async () => {
+      const modal = await openModal(page, '#sheet-modal');
+
+      await modal.callMethod('setCurrentBreakpoint', 0.25);
+      await page.waitForChanges();
+
+      const breakpoint = await modal.callMethod('getCurrentBreakpoint');
+      expect(breakpoint).toBe(0.25);
+    });
+
+  });
+
+  describe('setting the breakpoint to a valid value', () => {
+
+    it('should emit ionBreakpointDidChange', async () => {
+      const modal = await openModal(page, '#sheet-modal');
+
+      const ionBreakpointDidChangeSpy = await modal.spyOnEvent('ionBreakpointDidChange');
+
+      await modal.callMethod('setCurrentBreakpoint', 0.5);
+      await modal.waitForEvent('ionBreakpointDidChange');
+
+      expect(ionBreakpointDidChangeSpy).toHaveReceivedEventTimes(1);
+    });
+
+    it('should emit ionBreakpointDidChange when breakpoint is set to 0', async () => {
```

Comment:
Can we add a test to check that `ionBreakpointDidChange` is called when swiping down to breakpoint 0?